Fix lint rule and file violations

The current `make lint` produces the following:
```
flake8 kfp tests
kfp:0:1: E902 FileNotFoundError: [Errno 2] No such file or directory: 'kfp'
tests:0:1: E902 FileNotFoundError: [Errno 2] No such file or directory: 'tests'
make: *** [lint] Error 1
```
This rule should be updated to `flake8 kfp_notebook etc` and the subsequent 111 issues (at the time of this issue) should be resolved. 

We should also make `lint` a prerequisite for the `dist` target so that regular dev builds perform linting automatically.

EDIT: By using the same `setup.cfg` as used in elyra, the number of issues decreases dramatically - so let's add that update to the list.

--- a/Makefile
+++ b/Makefile
@@ -57,7 +57,7 @@ clean-test:
 	rm -fr .pytest_cache
 
 lint: ## check style with flake8
-	flake8 kfp tests
+	flake8 kfp_notebook etc
 
 test-dependencies:
 	@pip install -q -r test-requirements.txt
@@ -90,7 +90,7 @@ bdist:
 sdist:
 	python setup.py sdist
 
-dist: clean ## builds source and wheel package
+dist: clean lint ## builds source and wheel package
 	@make sdist
 	@make bdist
 	ls -l dist

--- a/etc/docker-scripts/bootstrapper.py
+++ b/etc/docker-scripts/bootstrapper.py
@@ -270,13 +270,13 @@ class OpUtil(object):
                           (package, current_packages[package]))
                 elif version.parse(ver) > version.parse(current_packages[package]):
                     print("Updating %s package from version %s to %s..." % (package, current_packages[package], ver))
-                    to_install_list.append(package+'=='+ver)
+                    to_install_list.append(package + '==' + ver)
                 elif version.parse(ver) < version.parse(current_packages[package]):
                     print("Newer %s package with version %s already installed. Skipping..." %
                           (package, current_packages[package]))
             else:
                 print("Package not found. Installing %s package with version %s..." % (package, ver))
-                to_install_list.append(package+'=='+ver)
+                to_install_list.append(package + '==' + ver)
 
         if to_install_list:
             if user_volume_path:

--- a/etc/tests/test_bootstrapper.py
+++ b/etc/tests/test_bootstrapper.py
@@ -99,7 +99,7 @@ def main_method_setup_execution(monkeypatch, s3_setup, tmpdir, argument_dict):
         for file in test_file_list:
             if file != 'test-notebookA-output.ipynb':
                 assert s3_setup.stat_object(bucket_name=argument_dict['cos-bucket'],
-                                            object_name="test-directory/"+file)
+                                            object_name="test-directory/" + file)
                 if file == "test-notebookA.html":
                     with open("test-notebookA.html") as html_file:
                         assert 'TEST_ENV_VAR1: test_env_var1' in html_file.read()
@@ -155,7 +155,7 @@ def test_main_method_with_dir_outputs(monkeypatch, s3_setup, tmpdir):
                      'cos-dependencies-archive': 'test-archive.tgz',
                      'filepath': 'etc/tests/resources/test-notebookA.ipynb',
                      'inputs': 'test-file.txt;test,file.txt',
-                     'outputs': 'test-file', # this is the directory that contains the outputs
+                     'outputs': 'test-file',  # this is the directory that contains the outputs
                      'user-volume-path': None}
     main_method_setup_execution(monkeypatch, s3_setup, tmpdir, argument_dict)
 
@@ -239,7 +239,7 @@ def test_package_installation(monkeypatch, virtualenv):
                     'ansiwrap': '0.8.4',
                     'packaging': '20.4',
                     'text-extensions-for-pandas':
-                    "git+https://github.com/akchinSTC/text-extensions-for-pandas@50d5a1688fb723b5dd8139761830d3419042fee5"
+                    "git+https://github.com/akchinSTC/text-extensions-for-pandas@50d5a1688fb723b5dd8139761830d3419042fee5"  # noqa: E501
                     }
 
     mocked_func = mock.Mock(return_value="default", side_effect=[elyra_dict, to_install_dict])
@@ -287,7 +287,7 @@ def test_package_installation_with_target_path(monkeypatch, virtualenv):
                     'ansiwrap': '0.8.4',
                     'packaging': '20.4',
                     'text-extensions-for-pandas':
-                    "git+https://github.com/akchinSTC/text-extensions-for-pandas@50d5a1688fb723b5dd8139761830d3419042fee5"
+                    "git+https://github.com/akchinSTC/text-extensions-for-pandas@50d5a1688fb723b5dd8139761830d3419042fee5"  # noqa: E501
                     }
 
     mocked_func = mock.Mock(return_value="default", side_effect=[elyra_dict, to_install_dict])
@@ -335,7 +335,7 @@ def test_fail_convert_notebook_to_html(tmpdir):
     with tmpdir.as_cwd():
         # Recent versions raising typeError due to #1130
         # https://github.com/jupyter/nbconvert/pull/1130
-        with pytest.raises( (TypeError, nbformat.validator.NotebookValidationError) ):
+        with pytest.raises((TypeError, nbformat.validator.NotebookValidationError)):
             bootstrapper.NotebookFileOp.convert_notebook_to_html(notebook_file, notebook_output_html_file)
 
 
@@ -353,11 +353,10 @@ def test_get_file_object_store(monkeypatch, s3_setup, tmpdir):
 
         op.get_file_from_object_storage(file_to_get)
         assert os.path.isfile(file_to_get)
-        assert hs.fileChecksum(file_to_get, "sha256") == hs.fileChecksum(current_directory+file_to_get, "sha256")
+        assert hs.fileChecksum(file_to_get, "sha256") == hs.fileChecksum(current_directory + file_to_get, "sha256")
 
 
 def test_fail_get_file_object_store(monkeypatch, s3_setup, tmpdir):
-    bucket_name = "test-bucket"
     file_to_get = "test-file.txt"
 
     with tmpdir.as_cwd():
@@ -377,11 +376,10 @@ def test_put_file_object_store(monkeypatch, s3_setup, tmpdir):
     with tmpdir.as_cwd():
         s3_setup.fget_object(bucket_name, file_to_put, file_to_put)
         assert os.path.isfile(file_to_put)
-        assert hs.fileChecksum(file_to_put, "sha256") == hs.fileChecksum(current_directory+file_to_put, "sha256")
+        assert hs.fileChecksum(file_to_put, "sha256") == hs.fileChecksum(current_directory + file_to_put, "sha256")
 
 
 def test_fail_invalid_filename_put_file_object_store(monkeypatch, s3_setup):
-    bucket_name = "test-bucket"
     file_to_put = "LICENSE_NOT_HERE"
 
     with pytest.raises(FileNotFoundError):
@@ -474,5 +472,4 @@ def test_requirements_file():
 def test_fail_requirements_file_bad_delimiter():
     bad_requirements_file = "etc/tests/resources/test-bad-requirements-elyra.txt"
     with pytest.raises(ValueError):
-        list_dict = bootstrapper.OpUtil.package_list_to_dict(bad_requirements_file)
-
+        bootstrapper.OpUtil.package_list_to_dict(bad_requirements_file)

--- a/kfp_notebook/pipeline/_notebook_op.py
+++ b/kfp_notebook/pipeline/_notebook_op.py
@@ -31,7 +31,7 @@ and have python3
 
 # Inputs and Outputs separator character.  If updated,
 # same-named variable in bootstrapper.py must be updated!
-INOUT_SEPARATOR= ';'
+INOUT_SEPARATOR = ';'
 
 KFP_NOTEBOOK_ORG = os.getenv("KFP_NOTEBOOK_ORG", "elyra-ai")
 KFP_NOTEBOOK_BRANCH = os.getenv("KFP_NOTEBOOK_BRANCH", "master" if 'dev' in __version__ else "v" + __version__)
@@ -47,7 +47,7 @@ class NotebookOp(ContainerOp):
                  cos_dependencies_archive: str,
                  pipeline_outputs: Optional[List[str]] = None,
                  pipeline_inputs: Optional[List[str]] = None,
-                 pipeline_envs: Optional[Dict[str,str]] = None,
+                 pipeline_envs: Optional[Dict[str, str]] = None,
                  requirements_url: str = None,
                  bootstrap_script_url: str = None,
                  emptydir_volume_size: str = None,
@@ -131,17 +131,16 @@ class NotebookOp(ContainerOp):
                                  '--cos-bucket {cos_bucket} '
                                  '--cos-directory "{cos_directory}" '
                                  '--cos-dependencies-archive "{cos_dependencies_archive}" '
-                                 '--file "{notebook}" '.format(
-                                    container_work_dir=self.container_work_dir,
-                                    bootscript_url=self.bootstrap_script_url,
-                                    reqs_url=self.requirements_url,
-                                    cos_endpoint=self.cos_endpoint,
-                                    cos_bucket=self.cos_bucket,
-                                    cos_directory=self.cos_directory,
-                                    cos_dependencies_archive=self.cos_dependencies_archive,
-                                    notebook=self.notebook,
-                                    python_user_lib_path_target=self.python_user_lib_path_target,
-                                    )
+                                 '--file "{notebook}" '
+                                 .format(container_work_dir=self.container_work_dir,
+                                         bootscript_url=self.bootstrap_script_url,
+                                         reqs_url=self.requirements_url,
+                                         cos_endpoint=self.cos_endpoint,
+                                         cos_bucket=self.cos_bucket,
+                                         cos_directory=self.cos_directory,
+                                         cos_dependencies_archive=self.cos_dependencies_archive,
+                                         notebook=self.notebook,
+                                         python_user_lib_path_target=self.python_user_lib_path_target)
                                  )
 
             if self.pipeline_inputs:
@@ -163,7 +162,7 @@ class NotebookOp(ContainerOp):
         # We must deal with the envs after the superclass initialization since these amend the
         # container attribute that isn't available until now.
         if self.pipeline_envs:
-            for key,value in self.pipeline_envs.items():  # Convert dict entries to format kfp needs
+            for key, value in self.pipeline_envs.items():  # Convert dict entries to format kfp needs
                 self.container.add_env_variable(V1EnvVar(name=key, value=value))
 
         # If crio volume size is found then assume kubeflow pipelines environment is using CRI-o as
@@ -182,7 +181,8 @@ class NotebookOp(ContainerOp):
                                                      value=self.python_user_lib_path + ':$PYTHONPATH'))
 
     def _get_file_name_with_extension(self, name, extension):
-        """ Simple function to construct a string filename
+        """
+        Simple function to construct a string filename
         Args:
             name: name of the file
             extension: extension to append to the name
@@ -200,6 +200,7 @@ class NotebookOp(ContainerOp):
         trimmed_artifact_list = []
         for artifact_name in pipeline_array:
             if INOUT_SEPARATOR in artifact_name:  # if INOUT_SEPARATOR is in name, throw since this is our separator
-                raise ValueError("Illegal character ({}) found in filename '{}'.".format(INOUT_SEPARATOR, artifact_name))
+                raise \
+                    ValueError("Illegal character ({}) found in filename '{}'.".format(INOUT_SEPARATOR, artifact_name))
             trimmed_artifact_list.append(artifact_name.strip())
         return INOUT_SEPARATOR.join(trimmed_artifact_list)

--- a/kfp_notebook/tests/test_notebook_op.py
+++ b/kfp_notebook/tests/test_notebook_op.py
@@ -137,8 +137,8 @@ def test_user_volume_size():
 
 @pytest.mark.skip(reason="not sure if we should even test this")
 def test_default_bootstrap_url(notebook_op):
-    assert notebook_op.bootstrap_script_url == 'https://raw.githubusercontent.com/elyra-ai/kfp-notebook/' \
-                                                'v0.9.1/etc/docker-scripts/bootstrapper.py'
+    assert notebook_op.bootstrap_script_url == \
+           'https://raw.githubusercontent.com/elyra-ai/kfp-notebook/v0.9.1/etc/docker-scripts/bootstrapper.py'
 
 
 def test_override_bootstrap_url():
@@ -155,8 +155,8 @@ def test_override_bootstrap_url():
 
 @pytest.mark.skip(reason="not sure if we should even test this")
 def test_default_requirements_url(notebook_op):
-    assert notebook_op.requirements_url == 'https://raw.githubusercontent.com/elyra-ai/' \
-                                            'kfp-notebook/v0.9.1/etc/requirements-elyra.txt'
+    assert notebook_op.requirements_url == \
+           'https://raw.githubusercontent.com/elyra-ai/kfp-notebook/v0.9.1/etc/requirements-elyra.txt'
 
 
 def test_override_requirements_url():

--- a/setup.cfg
+++ b/setup.cfg
@@ -21,6 +21,7 @@ universal = 0
 # References:
 # https://flake8.readthedocs.io/en/latest/user/configuration.html
 # https://flake8.readthedocs.io/en/latest/user/error-codes.html
+# https://docs.openstack.org/hacking/latest/user/hacking.html
 exclude = __init__.py
 ignore =
     # Import formatting
@@ -32,7 +33,17 @@ ignore =
     # Ambiguous variable names
     E741,
     # Allow breaks after binary operators
-    W504
+    W504,
+    # Include name with TODOs as in # TODO(yourname)
+    H101,
+    # Do not import more than one module per line
+    H301,
+    # Alphabetically order imports by the full module path
+    H306,
+    # Multi line docstrings should start without a leading new line
+    H404,
+    # Multi line docstrings should start with a one line summary followed by an empty line
+    H405
 max-line-length = 120
 
 [aliases]

